recaptcha_enterprise/snippets/passwordLeakAssessment.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This is a generated sample, using the typeless sample bot. Please
+// look for the source TypeScript sample (.ts) for modifications.
+'use strict';
+
+// sample-metadata:
+//   title: Detect password leak with reCAPTCHA
+//   description: Detect if the password credential has been compromised.
+//   usage: node passwordLeakAssessment.ts [project-id] [sita-key] [token] [action]
+
+const args = process.argv.slice(2);
+const [projectId, siteKey, token, action, username, password] = args;
+
+// [START recaptcha_enterprise_password_leak_verification]
+
+const {
+  RecaptchaEnterpriseServiceClient,
+} = require('@google-cloud/recaptcha-enterprise');
+const {PasswordCheckVerification} = require('recaptcha-password-check-helpers');
+
+// TODO(developer): Uncomment and set the following variables
+// Google Cloud Project ID.
+// const projectId: string = "PROJECT_ID"
+
+// Site key obtained by registering a domain/app to use recaptcha Enterprise.
+// const siteKey: string = "RECAPTCHA_SITE_KEY"
+
+// The token obtained from the client on passing the recaptchaSiteKey.
+// To get the token, integrate the recaptchaSiteKey with frontend. See,
+// https://cloud.google.com/recaptcha-enterprise/docs/instrument-web-pages#frontend_integration_score
+// const token: string = "TOKEN"
+
+// Action name corresponding to the token.
+// const action: string = "ACTION"
+
+// Username and password to be checked for credential breach.
+// const username: string = "user123";
+// const password: string = "1234@abcd";
+
+// Create a client.
+const client = new RecaptchaEnterpriseServiceClient();
+
+/*
+* Detect password leaks and breached credentials to prevent account takeovers
+* (ATOs) and credential stuffing attacks.
+* For more information, see:
+* https://cloud.google.com/recaptcha-enterprise/docs/getting-started and
+* https://security.googleblog.com/2019/02/protect-your-accounts-from-data.html
+
+* Steps:
+* 1. Use the 'create' method to hash and Encrypt the hashed username and
+* password.
+* 2. Send the hash prefix (2-byte) and the encrypted credentials to create
+* the assessment.(Hash prefix is used to partition the database.)
+* 3. Password leak assessment returns a database whose prefix matches the
```

Comment:
Maybe more accurate: returns a list of encrypted credential hashes to be compared with the decryption of the returned re-encrypted credentials